jenkins_node_scanner.py

```diff
@@ -231,7 +232,10 @@ def main():
                 write_output(args.output_file, node_info)
 
         logging.debug('Waiting %d seconds', args.period)
-        time.sleep(args.period)
+
+        if args.period:
+            time.sleep(args.period)
+        break
```

Comment:
This `break` is done unconditionally. There is no loop anymore.

Reply:
Right. 😅 Well, this PR is still technically WIP...